src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Debugger_InProc.cs

```diff
@@ -33,5 +56,105 @@ public void SetBreakPoint(string fileName, int lineNumber, int columnIndex)
+                    actionSucceeded = true;
+                }
+                catch (COMException ex)
+                {
+                    if ((uint)ex.ErrorCode != OperationNotSupportedHResult)
+                    {
+                        string message = string.Format(
+                            CultureInfo.InvariantCulture,
+                            "Failed to raise debugger command, an unexpected '{0}' was thrown with the HResult of '{1}'.",
+                            typeof(COMException),
+                            ex.ErrorCode);
+
+                        throw new Exception(message, ex);
+                    }
+
+                    actionSucceeded = false;
+                }
+
+                return actionSucceeded;
+            };
+
+            // Repeat the the command if "Operation Not Supported" is thrown.
+            if (!this.TryWaitFor(DebuggerCommandRetryTimeout, predict))
+            {
+                string message = string.Format(
+                    CultureInfo.InvariantCulture,
+                    "Failed to raise debugger command within '{0}' seconds.",
+                    DebuggerCommandRetryTimeout.TotalSeconds);
+
+                throw new Exception(message);
+            }
+        }
+
+        /// <summary>
+        /// Polls for the specified delegate to return true for the given timeout.
+        /// </summary>
+        /// <param name="timeout">Timeout to keep polling.</param>
+        /// <param name="predicate">Delegate to invoke.</param>
+        /// <returns>True if the delegate returned true when polled, otherwise false.</returns>
+        public bool TryWaitFor(TimeSpan timeout, Func<bool> predicate)
+        {
+            return this.TryWaitFor(timeout, DefaultPollingInterCallSleep, predicate);
+        }
+
+        /// <summary>
+        /// Polls for the specified delegate to return true for the given timeout.
+        /// </summary>
+        /// <param name="timeout">Timeout to keep polling.</param>
+        /// <param name="interval">Time to wait between polling.</param>
+        /// <param name="predicate">Delegate to invoke.</param>
+        /// <returns>
+        /// True if the delegate returned true when polled, otherwise false.
+        /// </returns>
+        private bool TryWaitFor(TimeSpan timeout, TimeSpan interval, Func<bool> predicate)
+        {
+            DateTime endTime = DateTime.UtcNow + this.ApplyTimeoutMultiplier(timeout);
+            bool validationDelegateSuccess = false;
+
+            while (DateTime.UtcNow < endTime)
+            {
+                // Note: we don't do this inline in the while() condition and return the result of 
+                // (DateTime.Now < startTime + timeout) as this could lead to cases where the validation
+                // delegate returned true, at the boundary of the valid time, and would return false
+                // when hitting the return statement. 
+                if (predicate.Invoke())
```

Comment:
>.Invoke [](start = 29, length = 7)

Just ```predicate()```

Reply:
cool. thanks!

---
In reply to: [136207625](https://github.com/dotnet/roslyn/pull/21824#discussion_r136207625) [](ancestors = 136207625)